### ISSUE TITLE
Community Hub Top contributor cards are not updated after profile display name or username are updated 

### ISSUE THREAD

**Issue by mirunacurtean** [mozilla/sumo-project#634]
**Environment:**
Stage
Note that did not manage to test this on Production due to not having a top contributor account to modify. 

**Platform:**
Windows

**Browsers tested on** 
Firefox 81.0
Chrome

**Preconditions:**
Sign in with an account that is either a superuser/moderator that can edit other accounts profiles or sign in with an account that has contributed enough (for example: question responses) to show up listed in the Community Hub Top Contributors. 

**Steps to reproduce:** 
1. Go to https://support.allizom.org/en-US/community. 
2. Scroll to Top Contributors section and select to See All contributors where the account to be edited is listed. (any account when signed in with superuser account) i.e. https://support.allizom.org/en-US/community/top-contributors/kb
3. Select to View Profile of account you want to edit :
-  when selecting own account redirect to My Profile will be performed
-  when selecting other user account with a superuser account redirect to account's profile page will be performed. The option to **Edit user profile** should be listed at the top of the profile page. 
4.  Through either of the two methods of editing change an account's display or username. 
5. Navigate back to https://support.allizom.org/en-US/community/top-contributors/questions and view the edited profile contributor card. 

**Expected result:**
The username or display name that was just updated should show the latest update on the user's contributor card. 

**Actual result:**
Previous username/ display name is still displayed instead. When accessing the View Profile link the difference between the username/display name and profile username/display name can be seen. 
ex: https://support.allizom.org/en-US/community/top-contributors/questions
https://support.allizom.org/en-US/user/test02

**Own account edit example:** 
![own account edit is not updated immediatly](https://user-images.githubusercontent.com/53306802/94112109-a4db6500-fe4d-11ea-9510-7a774ec8bf0f.gif)

**Other user account edit example:** 
![superuser edit is not updated immediatly](https://user-images.githubusercontent.com/53306802/94112100-9ee58400-fe4d-11ea-902f-f57128d18c61.gif)

**Comment by akatsoulas** [comment]
I believe this is now updated. These profiles should be coming from ElasticSearch and there is not a live update at the moment. The change needs some time in order to propagate. This won't be an issue with the new search. @mirunacurtean if this is the case I suggest that we should close this ticket. We are in the process of deprecating the current search and investing time to fix this issue probably is not worth it.

**Comment by mirunacurtean** [comment]
@akatsoulas You're right that top contributor cards get been updated after a while so I've edited the ticket to reflect that. I agree that there's no need to fix this anymore since it will be solved by search upgrade, however, the ticket should be left open to re-test and close when it can no longer be reproduced. I've changed the milestone for this ticket to reflect that.

**Comment by akatsoulas** [comment]
Removed the bug label since this was the initial design

**Comment by akatsoulas** [comment]
@mirunacurtean I am moving this to the QA column to verify that it's no longer reproducible. If this is the case we can close it since the code is already released

**Comment by mirunacurtean** [comment]
This still seems relevant. For areas such as https://support.allizom.org/en-US/community/top-contributors/kb, the profile card is not instantly (or after 3-5 minutes) updated. 

![profile not updated](https://user-images.githubusercontent.com/53306802/108995019-31ac7800-76a5-11eb-86c2-7133de7caea5.gif)


It's interesting that in the community search results https://support.allizom.org/en-US/community/search?q=mc6 the profile card does display the updated username/display name.

**Comment by LeoMcA** [comment]
Yeah, this makes sense: the KB and L10N top contributor pages are still aggressively cached SQL, whereas the AAQ top contributor page and community search are coming from Elastic which is updated in near-real-time.

**Comment by mirunacurtean** [comment]
Though about what to do with this ticket. @LeoMcA If we keep it open in the case that we can change the caching situation sometimes in the future, then it should be removed from the ES7 General release milestone and set back in the backlog. 
If not, this ticket can move on.

**Comment by LeoMcA** [comment]
I don't think we'll be reimplementing the KB top contributors stuff in ES anytime soon.

It should be possible to invalidate the cached SQL when a user changes their profile (or just cache the metrics piece, and load the user profiles every time - it shouldn't be a particularly expensive query - that's probably the best solution). However, that's not related to ES, so moving to KTLO.

**Comment by akatsoulas** [comment]
This is not a priority. Closing
